src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -1927,13 +1927,6 @@ HRESULT GetFunctionInfoInternal(LPCBYTE ip, EECodeInfo * pCodeInfo)
 
     if (ShouldAvoidHostCalls())
     {
-        ExecutionManager::ReaderLockHolder rlh(NoHostCalls);
```

Comment:
This ReaderLockHolder logic needs to stay in place. GetFunctionInfoInternal from the profiler api may be called while other threads are suspended via OS thread suspension apis, and may be used when alloc/free/new/delete are not permitted to be used. However, the technique you're using for locking looks like it will always successfully acquire, so you don't need to restore the Acquired() api, and call below.

Reply:
In the following perspective
https://github.com/dotnet/runtime/pull/68607#discussion_r866693016
https://github.com/dotnet/runtime/pull/68607#discussion_r866733481
it is possibble to stop writers here and pass NoHostCalls to the codeman by call below
//TODO